packages/auth-types/index.d.ts

```diff
@@ -246,7 +248,7 @@ export interface UserMetadata {
 
 export type Persistence = string;
 
-export interface OAuthCredential extends AuthCredential {
+export abstract class OAuthCredential extends AuthCredential {
```

Comment:
`OAuthCredential` is not really abstract.

Reply:
Changed to concrete class and made the constructor private.